# Web Scraping

In [319]:
#Importing the libraries
import requests
from bs4 import BeautifulSoup

In [320]:
#Collect page link
page = requests.get('https://www.espncricinfo.com/records/tournament/batting-most-runs-career/icc-cricket-world-cup-2023-24-15338')

In [321]:
#Create a BeautifulSoup object
soup = BeautifulSoup(page.text,'html.parser')

In [322]:
table= soup.find("table")
print(table.prettify())

<table class="ds-w-full ds-table ds-table-xs ds-table-auto ds-w-full ds-overflow-scroll ds-scrollbar-hide">
 <thead class="ds-bg-fill-content-alternate ds-text-left">
  <tr class="">
   <td class="ds-min-w-max">
    <div class="ds-popper-wrapper">
     <span class="ds-cursor-pointer">
      Player
     </span>
    </div>
   </td>
   <td class="ds-min-w-max ds-text-right">
    <div class="ds-popper-wrapper">
     <span class="ds-cursor-pointer">
      Span
     </span>
    </div>
   </td>
   <td class="ds-min-w-max ds-text-right">
    <div class="ds-popper-wrapper">
     <span class="ds-cursor-pointer">
      Mat
     </span>
    </div>
   </td>
   <td class="ds-min-w-max ds-text-right">
    <div class="ds-popper-wrapper">
     <span class="ds-cursor-pointer">
      Inns
     </span>
    </div>
   </td>
   <td class="ds-min-w-max ds-text-right">
    <div class="ds-popper-wrapper">
     <span class="ds-cursor-pointer">
      NO
     </span>
    </div>
   </td>
   <td class="ds-min-w-max 

In [323]:
headers= table.find_all("span")
print(headers)

[<span class="ds-cursor-pointer">Player</span>, <span class="ds-cursor-pointer">Span</span>, <span class="ds-cursor-pointer">Mat</span>, <span class="ds-cursor-pointer">Inns</span>, <span class="ds-cursor-pointer">NO</span>, <span class="ds-cursor-pointer"><strong>Runs</strong></span>, <span class="ds-cursor-pointer">HS</span>, <span class="ds-cursor-pointer">Ave</span>, <span class="ds-cursor-pointer">BF</span>, <span class="ds-cursor-pointer">SR</span>, <span class="ds-cursor-pointer">100</span>, <span class="ds-cursor-pointer">50</span>, <span class="ds-cursor-pointer">0</span>, <span class="ds-cursor-pointer">4s</span>, <span class="ds-cursor-pointer">6s</span>, <span class="ds-cursor-pointer"><a class="ds-inline-flex ds-items-start ds-leading-none" href="/cricketers/virat-kohli-253802" title="V Kohli (IND)"><span class="ds-text-tight-s ds-font-regular ds-text-typo-primary hover:ds-text-typo-primary-hover ds-block">V Kohli (IND)</span></a></span>, <span class="ds-text-tight-s ds-fo

In [324]:
data_entries=[]
for i in headers:
  data_entry= i.text
  data_entries.append(data_entry)

print(data_entries)

['Player', 'Span', 'Mat', 'Inns', 'NO', 'Runs', 'HS', 'Ave', 'BF', 'SR', '100', '50', '0', '4s', '6s', 'V Kohli (IND)', 'V Kohli (IND)', '2023-2023', '9', '9', '3', '594', '103*', '99.00', '671', '88.52', '2', '5', '1', '55', '7', 'Q de Kock (SA)', 'Q de Kock (SA)', '2023-2023', '9', '9', '0', '591', '174', '65.66', '541', '109.24', '4', '-', '-', '57', '21', 'R Ravindra (NZ)', 'R Ravindra (NZ)', '2023-2023', '9', '9', '1', '565', '123*', '70.62', '521', '108.44', '3', '2', '-', '52', '17', 'RG Sharma (IND)', 'RG Sharma (IND)', '2023-2023', '9', '9', '0', '503', '131', '55.88', '414', '121.49', '1', '3', '1', '58', '24', 'DA Warner (AUS)', 'DA Warner (AUS)', '2023-2023', '9', '9', '0', '499', '163', '55.44', '473', '105.49', '2', '2', '-', '48', '20', 'HE van der Dussen (SA)', 'HE van der Dussen (SA)', '2023-2023', '9', '9', '1', '442', '133', '55.25', '499', '88.57', '2', '2', '-', '39', '8', 'MR Marsh (AUS)', 'MR Marsh (AUS)', '2023-2023', '8', '8', '1', '426', '177*', '60.85', '389'

In [325]:
#Seperating the column headers
titles = data_entries[:15]
print(titles)

['Player', 'Span', 'Mat', 'Inns', 'NO', 'Runs', 'HS', 'Ave', 'BF', 'SR', '100', '50', '0', '4s', '6s']


In [326]:
rows = table.find_all("tr")
row_data_list = []  # Initialize an empty list to store the lists of row data

for row in rows[1:]:
    # Extracting text from each <td> element
    columns = row.find_all("td")
    row_data = [column.text.strip() for column in columns]

    # Append the row_data list to the row_data_list
    row_data_list.append(row_data)

# Now row_data_list is a list of lists with the extracted data from each row
row_data_list

[['V Kohli (IND)',
  '2023-2023',
  '9',
  '9',
  '3',
  '594',
  '103*',
  '99.00',
  '671',
  '88.52',
  '2',
  '5',
  '1',
  '55',
  '7'],
 ['Q de Kock (SA)',
  '2023-2023',
  '9',
  '9',
  '0',
  '591',
  '174',
  '65.66',
  '541',
  '109.24',
  '4',
  '-',
  '-',
  '57',
  '21'],
 ['R Ravindra (NZ)',
  '2023-2023',
  '9',
  '9',
  '1',
  '565',
  '123*',
  '70.62',
  '521',
  '108.44',
  '3',
  '2',
  '-',
  '52',
  '17'],
 ['RG Sharma (IND)',
  '2023-2023',
  '9',
  '9',
  '0',
  '503',
  '131',
  '55.88',
  '414',
  '121.49',
  '1',
  '3',
  '1',
  '58',
  '24'],
 ['DA Warner (AUS)',
  '2023-2023',
  '9',
  '9',
  '0',
  '499',
  '163',
  '55.44',
  '473',
  '105.49',
  '2',
  '2',
  '-',
  '48',
  '20'],
 ['HE van der Dussen (SA)',
  '2023-2023',
  '9',
  '9',
  '1',
  '442',
  '133',
  '55.25',
  '499',
  '88.57',
  '2',
  '2',
  '-',
  '39',
  '8'],
 ['MR Marsh (AUS)',
  '2023-2023',
  '8',
  '8',
  '1',
  '426',
  '177*',
  '60.85',
  '389',
  '109.51',
  '2',
  '1',
  '1',


In [327]:
#Dataframe created from the table data
df = pd.DataFrame(row_data_list, columns=titles)
df.head()

Player       Span Mat Inns NO Runs    HS    Ave   BF      SR 100  \
0    V Kohli (IND)  2023-2023   9    9  3  594  103*  99.00  671   88.52   2   
1   Q de Kock (SA)  2023-2023   9    9  0  591   174  65.66  541  109.24   4   
2  R Ravindra (NZ)  2023-2023   9    9  1  565  123*  70.62  521  108.44   3   
3  RG Sharma (IND)  2023-2023   9    9  0  503   131  55.88  414  121.49   1   
4  DA Warner (AUS)  2023-2023   9    9  0  499   163  55.44  473  105.49   2   

  50  0  4s  6s  
0  5  1  55   7  
1  -  -  57  21  
2  2  -  52  17  
3  3  1  58  24  
4  2  -  48  20

In [331]:
# Extract country information from parantheses
df['Country'] = df['Player'].str.extract(r'\((\w+)\)')

In [332]:
# Removing span column as the data is only for ongoing world cup
df = df.drop('Span', axis=1)

In [334]:
df.shape

(145, 15)

**DATASET DESCRIPTION**

**Shape**- 145 rows and 15 columns

**Columns**-

1) Player Name

2) No. of Matches played

3) No. of Innings Played

4) No. of times batsman remained NOT OUT

5) Runs Scored

6) High Score

7) Batting Average (Runs Scored / Number of times dismissed)

8) Balls Faced

9) Strike Rate ((Runs Scored / Balls Faced) * 100)

10) No of 100s scored

11) No of 50s scored

12) No of ducks(0s)

13) No of 4s

14) No of 6s

15) Country

PREPROCESSING

In [340]:
#Replacing all '-' in columns with '0's
df['100'] = df['100'].replace('-', 0)
df['50'] = df['50'].replace('-', 0)
df['0'] = df['0'].replace('-', 0)
df

Player Mat Inns NO   Runs    HS    Ave   BF      SR 100  \
0             V Kohli (IND)   9    9  3  594.0  103*  99.00  671   88.52   2   
1            Q de Kock (SA)   9    9  0  591.0   174  65.66  541  109.24   4   
2           R Ravindra (NZ)   9    9  1  565.0  123*  70.62  521  108.44   3   
3           RG Sharma (IND)   9    9  0  503.0   131  55.88  414  121.49   1   
4           DA Warner (AUS)   9    9  0  499.0   163  55.44  473  105.49   2   
..                      ...  ..  ... ..    ...   ...    ...  ...     ...  ..   
140      JR Hazlewood (AUS)   9    5  3    4.0     2   2.00    5   80.00   0   
141  Fazalhaq Farooqi (AFG)   6    3  2    2.0    2*   2.00    9   22.22   0   
142        LH Ferguson (NZ)   6    2  1    1.0     1   1.00    6   16.66   0   
143          AT Carey (AUS)   1    1  0    NaN     0   0.00    2    0.00   0   
144         Usama Mir (PAK)   4    1  0    NaN     0   0.00    3    0.00   0   

    50  0  4s  6s Country  
0    5  1  55   7     IND  
1    0  0  57  21      SA  
2    2  0  52  17      NZ  
3    3  1  58  24     IND  
4    2  0  48  20     AUS  
..  .. ..  ..  ..     ...  
140  0  1   0   0     AUS  
141  0  1   0   0     AFG  
142  0  0   0   0      NZ  
143  0  1   0   0     AUS  
144  0  1   0   0     PAK  

[145 rows x 15 columns]

In [345]:
#removing * from HS column
df['HS'] = df['HS'].str.replace('*', '', regex=False)
df.head()

Player Mat Inns NO   Runs   HS    Ave   BF      SR 100 50  0  4s  \
0    V Kohli (IND)   9    9  3  594.0  103  99.00  671   88.52   2  5  1  55   
1   Q de Kock (SA)   9    9  0  591.0  174  65.66  541  109.24   4  0  0  57   
2  R Ravindra (NZ)   9    9  1  565.0  123  70.62  521  108.44   3  2  0  52   
3  RG Sharma (IND)   9    9  0  503.0  131  55.88  414  121.49   1  3  1  58   
4  DA Warner (AUS)   9    9  0  499.0  163  55.44  473  105.49   2  2  0  48   

   6s Country  
0   7     IND  
1  21      SA  
2  17      NZ  
3  24     IND  
4  20     AUS

Selecting only the players with more than 250 runs from the dataframe and removing the others as only these players have a fighting chance to be the top scorer in the tournament.

In [346]:
df['Runs'] = pd.to_numeric(df['Runs'], errors='coerce')
min_runs= 250
new_df = df[df['Runs'] > min_runs]
new_df

Player Mat Inns NO   Runs   HS    Ave   BF      SR  \
0                V Kohli (IND)   9    9  3  594.0  103  99.00  671   88.52   
1               Q de Kock (SA)   9    9  0  591.0  174  65.66  541  109.24   
2              R Ravindra (NZ)   9    9  1  565.0  123  70.62  521  108.44   
3              RG Sharma (IND)   9    9  0  503.0  131  55.88  414  121.49   
4              DA Warner (AUS)   9    9  0  499.0  163  55.44  473  105.49   
5       HE van der Dussen (SA)   9    9  1  442.0  133  55.25  499   88.57   
6               MR Marsh (AUS)   8    8  1  426.0  177  60.85  389  109.51   
7                SS Iyer (IND)   9    9  3  421.0  128  70.16  395  106.58   
8             DJ Mitchell (NZ)   9    8  1  418.0  130  59.71  378  110.58   
9               DJ Malan (ENG)   9    9  0  404.0  140  44.88  400  101.00   
10            GJ Maxwell (AUS)   7    7  2  397.0  201  79.40  260  152.69   
11             AK Markram (SA)   9    9  1  396.0  106  49.50  346  114.45   
12       Mohammad Rizwan (PAK)   9    8  2  395.0  131  65.83  414   95.41   
13        Ibrahim Zadran (AFG)   9    9  1  376.0  129  47.00  493   76.26   
14       S Samarawickrama (SL)   9    9  2  373.0  108  53.28  365  102.19   
15              DP Conway (NZ)   9    9  1  359.0  152  44.87  351  102.27   
16    Azmatullah Omarzai (AFG)   9    8  3  353.0   97  70.60  361   97.78   
17              KL Rahul (IND)   9    8  3  347.0  102  69.40  371   93.53   
18     Abdullah Shafique (PAK)   8    8  0  336.0  113  42.00  360   93.33   
19             P Nissanka (SL)   9    9  1  332.0   77  41.50  371   89.48   
20           Mahmudullah (BAN)   8    7  1  328.0  111  54.66  358   91.62   
21              H Klaasen (SA)   9    9  1  326.0  109  40.75  232  140.51   
22            Babar Azam (PAK)   9    9  1  320.0   74  40.00  386   82.90   
23           Rahmat Shah (AFG)   9    9  1  320.0   77  40.00  419   76.37   
24  Hashmatullah Shahidi (AFG)   9    9  3  310.0   80  51.66  424   73.11   
25             BA Stokes (ENG)   6    6  0  304.0  108  50.66  341   89.14   
26        SA Engelbrecht (NED)   8    8  0  300.0   70  37.50  450   66.66   
27             BKG Mendis (SL)   9    9  0  294.0  122  32.66  258  113.95   
28           KIC Asalanka (SL)   9    8  0  288.0  108  36.00  338   85.20   
29         M Labuschagne (AUS)   9    8  0  286.0   71  35.75  371   77.08   
30            Litton Das (BAN)   9    9  0  284.0   76  31.55  354   80.22   
31    Rahmanullah Gurbaz (AFG)   9    9  0  280.0   80  31.11  283   98.93   
32               JE Root (ENG)   9    9  0  276.0   82  30.66  312   88.46   
33          Shubman Gill (IND)   7    7  0  270.0   92  38.57  258  104.65   
34             SPD Smith (AUS)   8    8  1  268.0   71  38.28  302   88.74   
35            SA Edwards (NED)   9    9  2  259.0   78  37.00  298   86.91   
36              DA Miller (SA)   9    9  2  255.0   53  36.42  214  119.15   

   100 50  0  4s  6s Country  
0    2  5  1  55   7     IND  
1    4  0  0  57  21      SA  
2    3  2  0  52  17      NZ  
3    1  3  1  58  24     IND  
4    2  2  0  48  20     AUS  
5    2  2  0  39   8      SA  
6    2  1  1  42  20     AUS  
7    1  3  1  32  16     IND  
8    1  2  0  39  15      NZ  
9    1  2  0  50   9     ENG  
10   2  0  1  40  22     AUS  
11   1  3  0  42   9      SA  
12   1  1  0  38   5     PAK  
13   1  1  0  39   5     AFG  
14   1  2  1  36   4      SL  
15   1  0  1  51   4      NZ  
16   0  3  0  26  13     AFG  
17   1  1  0  32   7     IND  
18   1  3  1  36   9     PAK  
19   0  4  2  44   3      SL  
20   1  1  0  28  14     BAN  
21   1  1  0  24  17      SA  
22   0  4  0  32   4     PAK  
23   0  3  0  28   2     AFG  
24   0  3  0  25   2     AFG  
25   1  2  1  26  11     ENG  
26   0  2  0  25   3     NED  
27   1  1  0  27  15      SL  
28   1  1  0  19  11      SL  
29   0  2  0  25   2     AUS  
30   0  2  1  32   5     BAN  
31   0  2  1  31   9     AFG  
32   0  3  1  21   2     ENG  
33   

Now keeping players only from Australia, New Zealand, India, and South Africa as they only have chance to be top scorer.

In [347]:
# List of countries to keep
countries = ['AUS', 'IND', 'NZ', 'SA']

# Filter the DataFrame based on the desired countries
new_df = new_df[new_df['Country'].isin(countries)]
new_df = new_df.reset_index(drop=True)

new_df

Player Mat Inns NO   Runs   HS    Ave   BF      SR 100 50  \
0            V Kohli (IND)   9    9  3  594.0  103  99.00  671   88.52   2  5   
1           Q de Kock (SA)   9    9  0  591.0  174  65.66  541  109.24   4  0   
2          R Ravindra (NZ)   9    9  1  565.0  123  70.62  521  108.44   3  2   
3          RG Sharma (IND)   9    9  0  503.0  131  55.88  414  121.49   1  3   
4          DA Warner (AUS)   9    9  0  499.0  163  55.44  473  105.49   2  2   
5   HE van der Dussen (SA)   9    9  1  442.0  133  55.25  499   88.57   2  2   
6           MR Marsh (AUS)   8    8  1  426.0  177  60.85  389  109.51   2  1   
7            SS Iyer (IND)   9    9  3  421.0  128  70.16  395  106.58   1  3   
8         DJ Mitchell (NZ)   9    8  1  418.0  130  59.71  378  110.58   1  2   
9         GJ Maxwell (AUS)   7    7  2  397.0  201  79.40  260  152.69   2  0   
10         AK Markram (SA)   9    9  1  396.0  106  49.50  346  114.45   1  3   
11          DP Conway (NZ)   9    9  1  359.0  152  44.87  351  102.27   1  0   
12          KL Rahul (IND)   9    8  3  347.0  102  69.40  371   93.53   1  1   
13          H Klaasen (SA)   9    9  1  326.0  109  40.75  232  140.51   1  1   
14     M Labuschagne (AUS)   9    8  0  286.0   71  35.75  371   77.08   0  2   
15      Shubman Gill (IND)   7    7  0  270.0   92  38.57  258  104.65   0  3   
16         SPD Smith (AUS)   8    8  1  268.0   71  38.28  302   88.74   0  2   
17          DA Miller (SA)   9    9  2  255.0   53  36.42  214  119.15   0  1   

    0  4s  6s Country  
0   1  55   7     IND  
1   0  57  21      SA  
2   0  52  17      NZ  
3   1  58  24     IND  
4   0  48  20     AUS  
5   0  39   8      SA  
6   1  42  20     AUS  
7   1  32  16     IND  
8   0  39  15      NZ  
9   1  40  22     AUS  
10  0  42   9      SA  
11  1  51   4      NZ  
12  0  32   7     IND  
13  0  24  17      SA  
14  0  25   2     AUS  
15  0  33   9     IND  
16  1  27   2     AUS  
17  0  17  15      SA

Only 18 players remain and only players from this list have a fighting chance to be the top run scorer of the ICC World Cup 2023.

Creating a new column FORM with the formula

Form = (0.4262 x Average) + (0.2566 x no. of innings) + (0.1510 x SR) + (0.0787 x Centuries) + (0.0556 x Fifties) + (0.0328 x HS)

This column helps predict the current form the batsman is in, in the tournament.
Formula is taken from a research paper by Kalpdrum Passi and Niravkumar Pandey. They have taken the form with stats over a year, instead I have taken the stats ranging only the tournament.

In [348]:
new_df['Ave'] = pd.to_numeric(new_df['Ave'], errors='coerce')
new_df['Inns'] = pd.to_numeric(new_df['Inns'], errors='coerce')
new_df['SR'] = pd.to_numeric(new_df['SR'], errors='coerce')
new_df['100'] = pd.to_numeric(new_df['100'], errors='coerce')
new_df['50'] = pd.to_numeric(new_df['50'], errors='coerce')
new_df['HS'] = pd.to_numeric(new_df['HS'], errors='coerce')

# Apply the formula
new_df['Form'] = (
    0.4262 * new_df['Ave'] +
    0.2566 * new_df['Inns'] +
    0.1510 * new_df['SR'] +
    0.0787 * new_df['100'] +
    0.0556 * new_df['50'] +
    0.0328 * new_df['HS']
)

In [349]:
new_df

Player Mat  Inns NO   Runs   HS    Ave   BF      SR  100  \
0            V Kohli (IND)   9     9  3  594.0  103  99.00  671   88.52    2   
1           Q de Kock (SA)   9     9  0  591.0  174  65.66  541  109.24    4   
2          R Ravindra (NZ)   9     9  1  565.0  123  70.62  521  108.44    3   
3          RG Sharma (IND)   9     9  0  503.0  131  55.88  414  121.49    1   
4          DA Warner (AUS)   9     9  0  499.0  163  55.44  473  105.49    2   
5   HE van der Dussen (SA)   9     9  1  442.0  133  55.25  499   88.57    2   
6           MR Marsh (AUS)   8     8  1  426.0  177  60.85  389  109.51    2   
7            SS Iyer (IND)   9     9  3  421.0  128  70.16  395  106.58    1   
8         DJ Mitchell (NZ)   9     8  1  418.0  130  59.71  378  110.58    1   
9         GJ Maxwell (AUS)   7     7  2  397.0  201  79.40  260  152.69    2   
10         AK Markram (SA)   9     9  1  396.0  106  49.50  346  114.45    1   
11          DP Conway (NZ)   9     9  1  359.0  152  44.87  351  102.27    1   
12          KL Rahul (IND)   9     8  3  347.0  102  69.40  371   93.53    1   
13          H Klaasen (SA)   9     9  1  326.0  109  40.75  232  140.51    1   
14     M Labuschagne (AUS)   9     8  0  286.0   71  35.75  371   77.08    0   
15      Shubman Gill (IND)   7     7  0  270.0   92  38.57  258  104.65    0   
16         SPD Smith (AUS)   8     8  1  268.0   71  38.28  302   88.74    0   
17          DA Miller (SA)   9     9  2  255.0   53  36.42  214  119.15    0   

    50  0  4s  6s Country       Form  
0    5  1  55   7     IND  61.683520  
1    0  0  57  21      SA  52.810932  
2    2  0  52  17      NZ  53.163784  
3    3  1  58  24     IND  49.012746  
4    2  0  48  20     AUS  47.481918  
5    2  0  39   8      SA  43.862020  
6    1  1  42  20     AUS  50.541680  
7    3  1  32  16     IND  52.749072  
8    2  0  39  15      NZ  48.652682  
9    0  1  40  22     AUS  65.442870  
10   3  0  42   9      SA  44.410550  
11   0  1  51   4      NZ  41.940064  
12   1  0  32   7     IND  49.234010  
13   1  0  24  17      SA  44.603560  
14   2  0  25   2     AUS  31.368530  
15   3  0  33   9     IND  37.221284  
16   2  1  27   2     AUS  34.207476  
17   1  0  17  15      SA  37.617254

# **PREDICTING RUNS OF TOP 5 BATSMEN**

# Predicted runs of Virat Kohli

In [416]:
df= pd.read_csv('/content/Virat_Kohli_data - Sheet1.csv')

In [417]:
df = df.drop(['Match','Innings', 'How Dismissed', 'Unnamed: 10', 'Aggr', 'Avg', 'S/R.1'], axis=1)
df = df.drop(['M/Inns','B/F','S/R'],axis=1)

In [418]:
from sklearn.preprocessing import LabelEncoder

# Create a LabelEncoder for 'Versus'
le_versus = LabelEncoder()
df['Versus'] = le_versus.fit_transform(df['Versus'])

# Print the labels and their corresponding classes for 'Versus'
print("Versus Labels:", le_versus.classes_)
print("Versus Encoded Values:", set(df['Versus']))

# Create a LabelEncoder for 'Ground'
le_ground = LabelEncoder()
df['Ground'] = le_ground.fit_transform(df['Ground'])

# Print the labels and their corresponding classes for 'Ground'
print("Ground Labels:", le_ground.classes_)
print("Ground Encoded Values:", set(df['Ground']))

Versus Labels: ['Afghanistan' 'Australia' 'Bangladesh' 'England' 'Ireland' 'Nepal'
 'Netherlands' 'New Zealand' 'Pakistan' 'South Africa' 'Sri Lanka'
 'United Arab Emirates' 'West Indies' 'Zimbabwe' nan]
Versus Encoded Values: {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14}
Ground Labels: ['Adelaide Oval' 'Arun Jaitley Stadium' 'Barabati Stadium'
 'Barsapara Cricket Stadium' 'Bay Oval' 'Bellerive Oval'
 'Bharat Ratna Shri Atal Bihari Vajpayee Ekana Cricket Stadium'
 'Boland Park' 'Brabourne Stadium' 'Brisbane Cricket Ground'
 'Captain Roop Singh Stadium' 'Dr YS Rajasekhara Reddy Cricket Stadium'
 'Eden Gardens' 'Eden Park' 'Edgbaston' 'Green Park'
 'Greenfield International Stadium' 'Harare Sports Club' 'Headingley'
 'Himachal Pradesh Cricket Association Stadium' 'Holkar Cricket Stadium'
 'JSCA International Stadium Complex' 'Kennington Oval' 'Kensington Oval'
 'Khan Shaheb Osman Ali Stadium' 'Kingsmead' "Lord's"
 'M Chinnaswamy Stadium' 'MA Chidambaram Stadium'
 'Madhavrao Scindia 

In [419]:
df['Date'] = pd.to_datetime(df['Date'])  # Convert 'Date' to datetime format

# Extract relevant date components
df['Year'] = df['Date'].dt.year
df['Month'] = df['Date'].dt.month
df['Day'] = df['Date'].dt.day


# Drop the original 'Date' column if needed
df = df.drop(['Date'], axis=1)

<ipython-input-419-bb0f0cec025a>:1: UserWarning: Parsing dates in DD/MM/YYYY format when dayfirst=False (the default) was specified. This may lead to inconsistently parsed dates! Specify a format to ensure consistent parsing.
  df['Date'] = pd.to_datetime(df['Date'])  # Convert 'Date' to datetime format


In [420]:
df = df[df['Runs'] != '-']

df['Runs'] = df['Runs'].str.replace('*', '')
df['Runs'] = df['Runs'].astype(float)


<ipython-input-420-20c42d6ad156>:3: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  df['Runs'] = df['Runs'].str.replace('*', '')
<ipython-input-420-20c42d6ad156>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['Runs'] = df['Runs'].str.replace('*', '')
<ipython-input-420-20c42d6ad156>:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['Runs'] = df['R

In [422]:
#Exchanging values of month and day in last 3 columns
df.loc[df.index[-3:], ['Month', 'Day']] = df.loc[df.index[-3:], ['Day', 'Month']].values

In [421]:
df = df.dropna()


In [423]:
df

Versus  Ground   Runs    Year  Month   Day
0        10      47   12.0  2008.0    8.0  18.0
1        10      47   37.0  2008.0    8.0  20.0
2        10      45   25.0  2008.0    8.0  24.0
3        10      45   54.0  2008.0    8.0  27.0
4        10      45   31.0  2008.0    8.0  29.0
..      ...     ...    ...     ...    ...   ...
285       7      19   95.0  2023.0   10.0  22.0
286       3       6    0.0  2023.0   10.0  29.0
287      10      67   88.0  2023.0   11.0   2.0
288       9      12  101.0  2023.0   11.0   5.0
289       6      27   51.0  2023.0   11.0  12.0

[278 rows x 6 columns]

In [424]:
df.shape

(278, 6)

In [438]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.preprocessing import StandardScaler

X = df.drop(['Runs'], axis=1)  # Features (exclude the target column)
y = df['Runs']  # Target variable

# Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Standardize the data
scaler = StandardScaler()
X_train_scaled = scaler.fit_transform(X_train)
X_test_scaled = scaler.transform(X_test)

# Create a simple neural network model
model = Sequential()
model.add(Dense(32, input_dim=X_train.shape[1], activation='relu'))
model.add(Dense(1, activation='linear'))

model.compile(optimizer='adam', loss='mean_squared_error', metrics=['mae'])

In [439]:
model.fit(X_train_scaled, y_train, epochs=100, verbose=1)


Epoch 1/100
7/7 [==============================] - 1s 3ms/step - loss: 4149.1670 - mae: 48.3360
Epoch 2/100
7/7 [==============================] - 0s 3ms/step - loss: 4138.0190 - mae: 48.2358
Epoch 3/100
7/7 [==============================] - 0s 3ms/step - loss: 4127.4746 - mae: 48.1393
Epoch 4/100
7/7 [==============================] - 0s 2ms/step - loss: 4116.6753 - mae: 48.0387
Epoch 5/100
7/7 [==============================] - 0s 2ms/step - loss: 4105.3477 - mae: 47.9397
Epoch 6/100
7/7 [==============================] - 0s 4ms/step - loss: 4093.9685 - mae: 47.8341
Epoch 7/100
7/7 [==============================] - 0s 2ms/step - loss: 4082.6399 - mae: 47.7380
Epoch 8/100
7/7 [==============================] - 0s 3ms/step - loss: 4070.9966 - mae: 47.6332
Epoch 9/100
7/7 [==============================] - 0s 3ms/step - loss: 4058.9314 - mae: 47.5261
Epoch 10/100
7/7 [==============================] - 0s 3ms/step - loss: 4046.9731 - mae: 47.4202
Epoch 11/100
7/7 [=====================

In [440]:
score = model.evaluate(X_test_scaled, y_test, verbose=0)
print(f'Mean Squared Error on test set: {score}')

Mean Squared Error on test set: [2616.732421875, 39.0604133605957]


In [441]:
# Now, let's use the model to make predictions for the semis
# Sample data for the semis (replace this with actual data)
semis_data = {'Versus': [7], 'Ground': [68], 'Year': [2023], 'Month': [11],'Day': [15]}
semis_df = pd.DataFrame(semis_data)
semis_df

Versus  Ground  Year  Month  Day
0       7      68  2023     11   15

7 is NZ in label encoder, and 68 is Wankhede Stadium, the venue of the semis.

In [442]:
# Standardize the semis data using the same scaler
semis_data_scaled = scaler.transform(semis_df)

In [443]:
# Make predictions
predicted_runs = model.predict(semis_data_scaled)

1/1 [==============================] - 0s 47ms/step


In [444]:
print(f'Predicted runs in the semis: {predicted_runs[0][0]:.2f}')

Predicted runs in the semis: 27.03


27 runs are Kohli's predicted runs in the semis.

In [447]:
# Now, let's use the model to make predictions for the semis
# Sample data for the semis (replace this with actual data)
finals_data = {'Versus': [1], 'Ground': [35], 'Year': [2023], 'Month': [11],'Day': [19]}
finals_df = pd.DataFrame(semis_data)
finals_df

Versus  Ground  Year  Month  Day
0       1      35  2023     11   19

35 represents Narendra Modi stadium, the venue of the final and 1 represents Australia, the predicted finalist(predicted in other task).

In [448]:
# Standardize the semis data using the same scaler
finals_data_scaled = scaler.transform(finals_df)

In [449]:
# Make predictions
predicted_runs = model.predict(finals_data_scaled)

1/1 [==============================] - 0s 146ms/step


In [450]:
print(f'Predicted runs in the finals: {predicted_runs[0][0]:.2f}')

Predicted runs in the finals: 37.85


38 runs are Kohli's predicted runs in the finals.

# Predicted semis runs of Q. de Kock

In [89]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [194]:
df = pd.read_csv('/content/de_Kock_Semis - Sheet1.csv')

In [195]:
df.head()

Match  Innings        Date M/Inns       Versus  \
0     1      1.0  19/01/2013    1st  New Zealand   
1     2      2.0  22/01/2013    2nd  New Zealand   
2     3      3.0  25/01/2013    2nd  New Zealand   
3     4      4.0  24/03/2013    2nd     Pakistan   
4     5      5.0  26/07/2013    1st    Sri Lanka   

                                    Ground  \
0                              Boland Park   
1                             Diamond Oval   
2                              Senwes Park   
3                         Willowmoore Park   
4  Pallekele International Cricket Stadium   

                        How Dismissed Runs   B/F    S/R  Unnamed: 10  Aggr  \
0     c N L McCullum b J E C Franklin   18  34.0  52.94          NaN  18.0   
1                         b K D Mills   25  28.0  89.29          NaN  43.0   
2                    b J E C Franklin   31  38.0  81.58          NaN  74.0   
3                       b Junaid Khan    3  12.0  25.00          NaN  77.0   
4  c †K C Sangakkara b N L T C Perera   20  28.0  71.43          NaN  97.0   

     Avg  S/R.1  
0  18.00  52.94  
1  21.50  69.35  
2  24.67  74.00  
3  19.25  68.75  
4  19.40  69.29

In [196]:
df = df.drop(['Match','Innings', 'How Dismissed', 'Unnamed: 10', 'Aggr', 'Avg', 'S/R.1'], axis=1)
df = df.drop(['M/Inns','B/F','S/R'],axis=1)

In [197]:
df.head()

Date       Versus                                   Ground Runs
0  19/01/2013  New Zealand                              Boland Park   18
1  22/01/2013  New Zealand                             Diamond Oval   25
2  25/01/2013  New Zealand                              Senwes Park   31
3  24/03/2013     Pakistan                         Willowmoore Park    3
4  26/07/2013    Sri Lanka  Pallekele International Cricket Stadium   20

Performing label encoding on categorical data.

In [198]:
from sklearn.preprocessing import LabelEncoder

# Create a LabelEncoder for 'Versus'
le_versus = LabelEncoder()
df['Versus'] = le_versus.fit_transform(df['Versus'])

# Print the labels and their corresponding classes for 'Versus'
print("Versus Labels:", le_versus.classes_)
print("Versus Encoded Values:", set(df['Versus']))

# Create a LabelEncoder for 'Ground'
le_ground = LabelEncoder()
df['Ground'] = le_ground.fit_transform(df['Ground'])

# Print the labels and their corresponding classes for 'Ground'
print("Ground Labels:", le_ground.classes_)
print("Ground Encoded Values:", set(df['Ground']))


Versus Labels: ['Afghanistan' 'Australia' 'Bangladesh' 'England' 'India' 'Ireland'
 'Netherlands' 'New Zealand' 'Pakistan' 'Sri Lanka' 'United Arab Emirates'
 'West Indies' 'Zimbabwe' nan]
Versus Encoded Values: {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13}
Ground Labels: ['Adelaide Oval' 'Arun Jaitley Stadium' 'Bay Oval' 'Bellerive Oval'
 'Bharat Ratna Shri Atal Bihari Vajpayee Ekana Cricket Stadium'
 'Boland Park' 'Buffalo Park' 'Diamond Oval' 'Eden Gardens' 'Eden Park'
 'Edgbaston' 'Green Park' 'Hagley Oval' 'Harare Sports Club' 'Headingley'
 'Himachal Pradesh Cricket Association Stadium' 'Holkar Cricket Stadium'
 'JSCA International Stadium Complex' 'Kennington Oval' 'Kensington Oval'
 'Kingsmead' "Lord's" 'MA Chidambaram Stadium'
 'Maharashtra Cricket Association Stadium'
 'Mahinda Rajapaksha International Cricket Stadium' 'Mangaung Oval'
 'Manuka Oval' 'Melbourne Cricket Ground' 'Narendra Modi Stadium'
 'Newlands' 'Old Trafford' 'Pallekele International Cricket Stadium'
 'Perth 

In [199]:
df.head()

Date  Versus  Ground Runs
0  19/01/2013       7       5   18
1  22/01/2013       7       7   25
2  25/01/2013       7      40   31
3  24/03/2013       8      55    3
4  26/07/2013       9      31   20

In [200]:
df['Date'] = pd.to_datetime(df['Date'])  # Convert 'Date' to datetime format

# Extract relevant date components
df['Year'] = df['Date'].dt.year
df['Month'] = df['Date'].dt.month
df['Day'] = df['Date'].dt.day

# Drop the original 'Date' column if needed
df['Date'] = pd.to_datetime(df['Date'], errors='coerce')


<ipython-input-200-33661fda80a0>:1: UserWarning: Parsing dates in DD/MM/YYYY format when dayfirst=False (the default) was specified. This may lead to inconsistently parsed dates! Specify a format to ensure consistent parsing.
  df['Date'] = pd.to_datetime(df['Date'])  # Convert 'Date' to datetime format


In [167]:
df['Runs'] = df['Runs'].str.replace('*', '')
df['Runs'] = df['Runs'].astype(float)


<ipython-input-167-cc59527166e4>:1: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  df['Runs'] = df['Runs'].str.replace('*', '')


In [168]:
df.shape

(155, 6)

In [169]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 155 entries, 0 to 154
Data columns (total 6 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Versus  155 non-null    int64  
 1   Ground  155 non-null    int64  
 2   Runs    154 non-null    float64
 3   Year    154 non-null    float64
 4   Month   154 non-null    float64
 5   Day     154 non-null    float64
dtypes: float64(4), int64(2)
memory usage: 7.4 KB


In [170]:
df

Versus  Ground   Runs    Year  Month   Day
0         7       5   18.0  2013.0    1.0  19.0
1         7       7   25.0  2013.0    1.0  22.0
2         7      40   31.0  2013.0    1.0  25.0
3         8      55    3.0  2013.0    3.0  24.0
4         9      31   20.0  2013.0    7.0  26.0
..      ...     ...    ...     ...    ...   ...
150       8      22   24.0  2023.0   10.0  27.0
151       7      23  114.0  2023.0    1.0  11.0
152       4       8    5.0  2023.0    5.0  11.0
153       0      28   41.0  2023.0   10.0  11.0
154      13      57    NaN     NaN    NaN   NaN

[155 rows x 6 columns]

In [172]:
nan= df.isna().sum()
nan

Versus    0
Ground    0
Runs      1
Year      1
Month     1
Day       1
dtype: int64

In [173]:
df = df.dropna()



In [174]:
df

Versus  Ground   Runs    Year  Month   Day
0         7       5   18.0  2013.0    1.0  19.0
1         7       7   25.0  2013.0    1.0  22.0
2         7      40   31.0  2013.0    1.0  25.0
3         8      55    3.0  2013.0    3.0  24.0
4         9      31   20.0  2013.0    7.0  26.0
..      ...     ...    ...     ...    ...   ...
149       2      53  174.0  2023.0   10.0  24.0
150       8      22   24.0  2023.0   10.0  27.0
151       7      23  114.0  2023.0    1.0  11.0
152       4       8    5.0  2023.0    5.0  11.0
153       0      28   41.0  2023.0   10.0  11.0

[154 rows x 6 columns]

In [176]:
#Exchanging values of month and day in last 3 columns
df.loc[df.index[-3:], ['Month', 'Day']] = df.loc[df.index[-3:], ['Day', 'Month']].values


In [177]:
df

Versus  Ground   Runs    Year  Month   Day
0         7       5   18.0  2013.0    1.0  19.0
1         7       7   25.0  2013.0    1.0  22.0
2         7      40   31.0  2013.0    1.0  25.0
3         8      55    3.0  2013.0    3.0  24.0
4         9      31   20.0  2013.0    7.0  26.0
..      ...     ...    ...     ...    ...   ...
149       2      53  174.0  2023.0   10.0  24.0
150       8      22   24.0  2023.0   10.0  27.0
151       7      23  114.0  2023.0   11.0   1.0
152       4       8    5.0  2023.0   11.0   5.0
153       0      28   41.0  2023.0   11.0  10.0

[154 rows x 6 columns]

In [178]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.preprocessing import StandardScaler

X = df.drop(['Runs'], axis=1)  # Features (exclude the target column)
y = df['Runs']  # Target variable

In [179]:
# Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

In [180]:
# Standardize the data
scaler = StandardScaler()
X_train_scaled = scaler.fit_transform(X_train)
X_test_scaled = scaler.transform(X_test)

In [181]:
from keras.models import Sequential
from keras.layers import Dense

In [182]:
# Create a simple neural network model
model = Sequential()
model.add(Dense(64, input_dim=X_train.shape[1], activation='relu'))
model.add(Dense(32, activation='relu'))
model.add(Dense(1, activation='linear'))

In [183]:
model.compile(optimizer='adam', loss='mean_squared_error', metrics=['mae'])


In [184]:
model.fit(X_train_scaled, y_train, epochs=100, batch_size=32, verbose=1)


Epoch 1/100
4/4 [==============================] - 1s 4ms/step - loss: 3700.7271 - mae: 44.8948
Epoch 2/100
4/4 [==============================] - 0s 3ms/step - loss: 3681.8484 - mae: 44.7034
Epoch 3/100
4/4 [==============================] - 0s 4ms/step - loss: 3663.0645 - mae: 44.5027
Epoch 4/100
4/4 [==============================] - 0s 4ms/step - loss: 3644.2739 - mae: 44.2972
Epoch 5/100
4/4 [==============================] - 0s 4ms/step - loss: 3623.1650 - mae: 44.0889
Epoch 6/100
4/4 [==============================] - 0s 4ms/step - loss: 3601.1787 - mae: 43.8590
Epoch 7/100
4/4 [==============================] - 0s 6ms/step - loss: 3578.6790 - mae: 43.6100
Epoch 8/100
4/4 [==============================] - 0s 4ms/step - loss: 3551.6255 - mae: 43.3471
Epoch 9/100
4/4 [==============================] - 0s 5ms/step - loss: 3524.4023 - mae: 43.0611
Epoch 10/100
4/4 [==============================] - 0s 3ms/step - loss: 3495.4670 - mae: 42.7569
Epoch 11/100
4/4 [=====================

In [185]:
score = model.evaluate(X_test_scaled, y_test, verbose=0)
print(f'Mean Squared Error on test set: {score}')

Mean Squared Error on test set: [2220.993408203125, 39.3176155090332]


In [201]:
# Now, let's use the model to make predictions for the semis
# Sample data for the semis (replace this with actual data)
semis_data = {'Versus': [1], 'Ground': [8], 'Year': [2023], 'Month': [11],'Day': [16]}
semis_df = pd.DataFrame(semis_data)
semis_df

Versus  Ground  Year  Month  Day
0       1       8  2023     11   16

Ground 8 is Eden Gardens, where semis are going to be played and Versus-1 represents Australia.

In [202]:
# Standardize the semis data using the same scaler
semis_data_scaled = scaler.transform(semis_df)

In [203]:
# Make predictions
predicted_runs = model.predict(semis_data_scaled)

1/1 [==============================] - 0s 83ms/step


In [204]:
print(f'Predicted runs in the semis: {predicted_runs[0][0]:.2f}')

Predicted runs in the semis: 56.47


**Predicted runs of Quinton de Cock are 56.**

# Predicted semis runs of Rachin Ravindra

In [297]:
df2 = pd.read_csv('/content/Rachin_Ravindra_data.csv')

In [298]:
df2.head()

Match  Innings        Date M/Inns     Versus                      Ground  \
0      1      1.0  25/03/2023    1st  Sri Lanka                   Eden Park   
1      2      NaN  31/03/2023    2nd  Sri Lanka                 Seddon Park   
2      3      2.0  27/04/2023    1st   Pakistan  Rawalpindi Cricket Stadium   
3      4      NaN  29/04/2023    1st   Pakistan  Rawalpindi Cricket Stadium   
4      5      3.0  07/05/2023    1st   Pakistan  National Stadium (Karachi)   

                    How Dismissed Runs B/F    S/R  Unnamed: 10  Aggr    Avg  \
0         c D Shanaka b K Rajitha   49  52  94.23          NaN    49  49.00   
1                     did not bat    -   -      -          NaN    49  49.00   
2  c Mohammad Nawaz b Naseem Shah    9   9    100          NaN    58  29.00   
3                     did not bat    -   -      -          NaN    58  29.00   
4       c Babar Azam b Haris Rauf   28  20    140          NaN    86  28.67   

    S/R.1  
0   94.23  
1   94.23  
2   95.08  
3   95.08  
4  106.17

In [299]:
df2 = df2.drop(['Match','Innings', 'How Dismissed', 'Unnamed: 10', 'Aggr', 'Avg', 'S/R.1'], axis=1)
df2 = df2.drop(['M/Inns','B/F','S/R'],axis=1)

In [300]:
from sklearn.preprocessing import LabelEncoder

# Create a LabelEncoder for 'Versus'
le_versus = LabelEncoder()
df2['Versus'] = le_versus.fit_transform(df2['Versus'])

# Print the labels and their corresponding classes for 'Versus'
print("Versus Labels:", le_versus.classes_)
print("Versus Encoded Values:", set(df2['Versus']))

# Create a LabelEncoder for 'Ground'
le_ground = LabelEncoder()
df2['Ground'] = le_ground.fit_transform(df2['Ground'])

# Print the labels and their corresponding classes for 'Ground'
print("Ground Labels:", le_ground.classes_)
print("Ground Encoded Values:", set(df2['Ground']))

Versus Labels: ['Afghanistan' 'Australia' 'Bangladesh' 'England' 'India' 'Netherlands'
 'Pakistan' 'South Africa' 'Sri Lanka']
Versus Encoded Values: {0, 1, 2, 3, 4, 5, 6, 7, 8}
Ground Labels: ['Eden Park' 'Himachal Pradesh Cricket Association Stadium'
 'Kennington Oval' "Lord's" 'M Chinnaswamy Stadium'
 'MA Chidambaram Stadium' 'Maharashtra Cricket Association Stadium'
 'Narendra Modi Stadium' 'National Stadium (Karachi)'
 'Rajiv Gandhi International Stadium' 'Rawalpindi Cricket Stadium'
 'Seddon Park' 'Shere Bangla National Stadium' 'Sophia Gardens'
 'The Rose Bowl']
Ground Encoded Values: {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14}


In [302]:
df2['Date'] = pd.to_datetime(df2['Date'], format='%d/%m/%Y', errors='coerce')



In [303]:
df2['Date'] = pd.to_datetime(df2['Date'])  # Convert 'Date' to datetime format

# Extract relevant date components
df2['Year'] = df2['Date'].dt.year
df2['Month'] = df2['Date'].dt.month
df2['Day'] = df2['Date'].dt.day

# Drop the original 'Date' column if needed
df2 = df2.drop(['Date'], axis=1)

In [305]:
df2 = df2[df2['Runs'] != '-']


In [306]:
df2['Runs'] = df2['Runs'].str.replace('*', '')
df2['Runs'] = df2['Runs'].astype(float)

<ipython-input-306-208e9bf742c1>:1: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  df2['Runs'] = df2['Runs'].str.replace('*', '')


In [307]:
#Exchanging values of month and day in last 3 columns
df2.loc[df2.index[-3:], ['Month', 'Day']] = df2.loc[df2.index[-3:], ['Day', 'Month']].values

In [308]:
df2

Versus  Ground   Runs  Year  Month  Day
0        8       0   49.0  2023      3   25
2        6      10    9.0  2023      4   27
4        6       8   28.0  2023      7    5
6        3      14    4.0  2023     10    9
7        3       2   28.0  2023      9   13
8        3       3   61.0  2023      9   15
9        2      12    0.0  2023      9   21
10       2      12   10.0  2023      9   23
12       3       7  123.0  2023      5   10
13       5       9   51.0  2023      9   10
14       2       5    9.0  2023     10   13
15       0       5   32.0  2023     10   18
16       4       1   75.0  2023     10   22
17       1       1  116.0  2023     10   28
18       7       6    9.0  2023     11    1
19       6       4  108.0  2023     11    4
20       8       4   42.0  2023     11    9

In [310]:
df2.shape

(17, 6)

In [311]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.preprocessing import StandardScaler

X = df2.drop(['Runs'], axis=1)  # Features (exclude the target column)
y = df2['Runs']  # Target variable

# Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Standardize the data
scaler = StandardScaler()
X_train_scaled = scaler.fit_transform(X_train)
X_test_scaled = scaler.transform(X_test)

# Create a simple neural network model
model = Sequential()
model.add(Dense(32, input_dim=X_train.shape[1], activation='relu'))
model.add(Dense(1, activation='linear'))

model.compile(optimizer='adam', loss='mean_squared_error', metrics=['mae'])


In [312]:
model.fit(X_train_scaled, y_train, epochs=100, verbose=1)


Epoch 1/100
1/1 [==============================] - 1s 546ms/step - loss: 3196.5962 - mae: 40.8231
Epoch 2/100
1/1 [==============================] - 0s 12ms/step - loss: 3195.0972 - mae: 40.8084
Epoch 3/100
1/1 [==============================] - 0s 12ms/step - loss: 3193.5977 - mae: 40.7937
Epoch 4/100
1/1 [==============================] - 0s 14ms/step - loss: 3192.0979 - mae: 40.7791
Epoch 5/100
1/1 [==============================] - 0s 12ms/step - loss: 3190.5989 - mae: 40.7643
Epoch 6/100
1/1 [==============================] - 0s 10ms/step - loss: 3189.0974 - mae: 40.7496
Epoch 7/100
1/1 [==============================] - 0s 15ms/step - loss: 3187.5962 - mae: 40.7348
Epoch 8/100
1/1 [==============================] - 0s 10ms/step - loss: 3186.0959 - mae: 40.7201
Epoch 9/100
1/1 [==============================] - 0s 11ms/step - loss: 3184.5959 - mae: 40.7053
Epoch 10/100
1/1 [==============================] - 0s 10ms/step - loss: 3183.0938 - mae: 40.6905
Epoch 11/100
1/1 [==========

In [313]:
score = model.evaluate(X_test_scaled, y_test, verbose=0)
print(f'Mean Squared Error on test set: {score}')

Mean Squared Error on test set: [4333.2939453125, 55.34046936035156]


In [314]:
# Now, let's use the model to make predictions for the semis
# Sample data for the semis (replace this with actual data)
semis_data = {'Versus': [4], 'Ground': [6], 'Year': [2023], 'Month': [11],'Day': [15]}
semis_df = pd.DataFrame(semis_data)
semis_df

Versus  Ground  Year  Month  Day
0       4       6  2023     11   15

Versus-4 represents India and as Wankhade stadium is not there, MCA stadium is closest in conditions to Wankhade, hence took 6 in ground.

In [315]:
# Standardize the semis data using the same scaler
semis_data_scaled = scaler.transform(semis_df)

In [317]:
# Make predictions
predicted_runs = model.predict(semis_data_scaled)

1/1 [==============================] - 0s 23ms/step


In [318]:
print(f'Predicted runs in the semis: {predicted_runs[0][0]:.2f}')

Predicted runs in the semis: 0.68


Predicted Runs of Rachin Ravendra are 1.

# Predicted Runs of Rohit Sharma.

In [350]:
df= pd.read_csv('/content/Rohit_Sharma_data - Sheet1.csv')

In [351]:
df = df.drop(['Match','Innings', 'How Dismissed', 'Unnamed: 10', 'Aggr', 'Avg', 'S/R.1'], axis=1)
df = df.drop(['M/Inns','B/F','S/R'],axis=1)

In [352]:
from sklearn.preprocessing import LabelEncoder

# Create a LabelEncoder for 'Versus'
le_versus = LabelEncoder()
df['Versus'] = le_versus.fit_transform(df['Versus'])

# Print the labels and their corresponding classes for 'Versus'
print("Versus Labels:", le_versus.classes_)
print("Versus Encoded Values:", set(df['Versus']))

# Create a LabelEncoder for 'Ground'
le_ground = LabelEncoder()
df['Ground'] = le_ground.fit_transform(df['Ground'])

# Print the labels and their corresponding classes for 'Ground'
print("Ground Labels:", le_ground.classes_)
print("Ground Encoded Values:", set(df['Ground']))


Versus Labels: ['Afghanistan' 'Australia' 'Bangladesh' 'England' 'Hong Kong' 'Ireland'
 'Nepal' 'Netherlands' 'New Zealand' 'Pakistan' 'South Africa' 'Sri Lanka'
 'United Arab Emirates' 'West Indies' 'Zimbabwe']
Versus Encoded Values: {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14}
Ground Labels: ['Adelaide Oval' 'Arun Jaitley Stadium' 'Barabati Stadium'
 'Barsapara Cricket Stadium' 'Bay Oval' 'Bellerive Oval'
 'Bharat Ratna Shri Atal Bihari Vajpayee Ekana Cricket Stadium'
 'Brabourne Stadium' 'Brisbane Cricket Ground'
 'Civil Service Cricket Club' 'Daren Sammy National Cricket Stadium'
 'Dr YS Rajasekhara Reddy Cricket Stadium'
 'Dubai International Cricket Stadium' 'Eden Gardens' 'Eden Park'
 'Edgbaston' 'Green Park' 'Greenfield International Stadium'
 'Harare Sports Club' 'Headingley'
 'Himachal Pradesh Cricket Association Stadium' 'Holkar Cricket Stadium'
 'JSCA International Stadium Complex' 'Kennington Oval' 'Kensington Oval'
 'Khan Shaheb Osman Ali Stadium' 'Kingsmead' "Lord'

In [354]:
df['Date'] = pd.to_datetime(df['Date'])  # Convert 'Date' to datetime format

# Extract relevant date components
df['Year'] = df['Date'].dt.year
df['Month'] = df['Date'].dt.month
df['Day'] = df['Date'].dt.day

# Drop the original 'Date' column if needed
df['Date'] = pd.to_datetime(df['Date'], errors='coerce')

In [355]:
df['Date'] = pd.to_datetime(df['Date'])  # Convert 'Date' to datetime format

# Extract relevant date components
df['Year'] = df['Date'].dt.year
df['Month'] = df['Date'].dt.month
df['Day'] = df['Date'].dt.day

# Drop the original 'Date' column if needed
df = df.drop(['Date'], axis=1)

In [357]:
df = df[df['Runs'] != '-']

df['Runs'] = df['Runs'].str.replace('*', '')
df['Runs'] = df['Runs'].astype(float)

<ipython-input-357-20c42d6ad156>:3: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  df['Runs'] = df['Runs'].str.replace('*', '')
<ipython-input-357-20c42d6ad156>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['Runs'] = df['Runs'].str.replace('*', '')
<ipython-input-357-20c42d6ad156>:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['Runs'] = df['R

In [361]:
#Exchanging values of month and day in last 3 columns
df.loc[df.index[-3:], ['Month', 'Day']] = df.loc[df.index[-3:], ['Day', 'Month']].values

In [362]:
df

Versus  Ground  Runs  Year  Month  Day
1        10       9   8.0  2007      6   26
2         1      47   1.0  2007      5   10
3         9      52  52.0  2007     11   18
4         1       8  29.0  2008      3    2
5        11       8   0.0  2008      4    2
..      ...     ...   ...   ...    ...  ...
255       8      20  46.0  2023     10   22
256       3       6  87.0  2023     10   29
257      11      67   4.0  2023     11    2
258      10      13  40.0  2023     11    5
259       7      28  61.0  2023     11   12

[252 rows x 6 columns]

In [359]:
df.shape

(252, 6)

In [360]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.preprocessing import StandardScaler

X = df.drop(['Runs'], axis=1)  # Features (exclude the target column)
y = df['Runs']  # Target variable

# Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Standardize the data
scaler = StandardScaler()
X_train_scaled = scaler.fit_transform(X_train)
X_test_scaled = scaler.transform(X_test)

# Create a simple neural network model
model = Sequential()
model.add(Dense(32, input_dim=X_train.shape[1], activation='relu'))
model.add(Dense(1, activation='linear'))

model.compile(optimizer='adam', loss='mean_squared_error', metrics=['mae'])


In [363]:
model.fit(X_train_scaled, y_train, epochs=100, verbose=1)


Epoch 1/100
7/7 [==============================] - 1s 6ms/step - loss: 4028.9045 - mae: 41.9871
Epoch 2/100
7/7 [==============================] - 0s 4ms/step - loss: 4019.7275 - mae: 41.8981
Epoch 3/100
7/7 [==============================] - 0s 4ms/step - loss: 4011.3887 - mae: 41.8131
Epoch 4/100
7/7 [==============================] - 0s 3ms/step - loss: 4002.4253 - mae: 41.7253
Epoch 5/100
7/7 [==============================] - 0s 3ms/step - loss: 3993.6577 - mae: 41.6402
Epoch 6/100
7/7 [==============================] - 0s 3ms/step - loss: 3985.0769 - mae: 41.5552
Epoch 7/100
7/7 [==============================] - 0s 3ms/step - loss: 3975.7793 - mae: 41.4746
Epoch 8/100
7/7 [==============================] - 0s 4ms/step - loss: 3967.2617 - mae: 41.3896
Epoch 9/100
7/7 [==============================] - 0s 4ms/step - loss: 3958.4626 - mae: 41.3051
Epoch 10/100
7/7 [==============================] - 0s 5ms/step - loss: 3949.1262 - mae: 41.2207
Epoch 11/100
7/7 [=====================

In [364]:
score = model.evaluate(X_test_scaled, y_test, verbose=0)
print(f'Mean Squared Error on test set: {score}')

Mean Squared Error on test set: [1625.9998779296875, 29.073476791381836]


In [365]:
# Now, let's use the model to make predictions for the semis
# Sample data for the semis (replace this with actual data)
semis_data = {'Versus': [8], 'Ground': [67], 'Year': [2023], 'Month': [11],'Day': [15]}
semis_df = pd.DataFrame(semis_data)
semis_df

Versus  Ground  Year  Month  Day
0       8      67  2023     11   15

In [366]:
# Standardize the semis data using the same scaler
semis_data_scaled = scaler.transform(semis_df)

In [367]:
# Make predictions
predicted_runs = model.predict(semis_data_scaled)

1/1 [==============================] - 0s 79ms/step


In [368]:
print(f'Predicted runs in the semis: {predicted_runs[0][0]:.2f}')

Predicted runs in the semis: 39.53


Rohit Sharma predicted runs in semis- 40

We are assuming INDIA vs AUSTRALIA final as per our other predictions.

In [370]:
# Now, let's use the model to make predictions for the semis
# Sample data for the semis (replace this with actual data)
finals_data = {'Versus': [1], 'Ground': [35], 'Year': [2023], 'Month': [11],'Day': [19]}
finals_df = pd.DataFrame(semis_data)
finals_df

Versus  Ground  Year  Month  Day
0       1      35  2023     11   19

In [371]:
# Standardize the semis data using the same scaler
finals_data_scaled = scaler.transform(finals_df)

In [372]:
# Make predictions
predicted_runs = model.predict(finals_data_scaled)

1/1 [==============================] - 0s 105ms/step


In [373]:
print(f'Predicted runs in the semis: {predicted_runs[0][0]:.2f}')

Predicted runs in the semis: 44.48


Rohit Sharma predicted runs in finals- 44.